vtkext/private/module/F3DColoringInfoHandler.cxx

```diff
@@ -0,0 +1,187 @@
+#include "F3DColoringInfoHandler.h"
+
+#include "F3DLog.h"
+
+#include <vtkCellData.h>
+#include <vtkDataArray.h>
+#include <vtkDataSet.h>
+#include <vtkPointData.h>
+
+#include <cassert>
+#include <set>
+
+//----------------------------------------------------------------------------
+void F3DColoringInfoHandler::ClearColoringInfo()
+{
+  this->PointDataColoringInfo.clear();
+  this->CellDataColoringInfo.clear();
+}
+
+//----------------------------------------------------------------------------
+void F3DColoringInfoHandler::UpdateColoringInfo(vtkDataSet* dataset, bool useCellData)
+{
+  // XXX: This assumes importer do not import actors with an empty input
+  assert(dataset);
+
+  // Recover all possible names
+  std::set<std::string> arrayNames;
+
+  vtkDataSetAttributes* attr = useCellData
+    ? static_cast<vtkDataSetAttributes*>(dataset->GetCellData())
+    : static_cast<vtkDataSetAttributes*>(dataset->GetPointData());
```

Comment:
Do you really need to cast since it's a base class for both?

Reply:
I do, without it compiler complains that they are not the same type.

Reply:
Ok I remember I already encounter something like that before.